### ISSUE TITLE
Renaming a class by its constructor doesn't work

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#4220]
Pressing `F2` to rename on a constructor like this doesn't do anything when you press `<enter>` (no error, no rename).

<img width="707" alt="Screenshot 2022-10-15 at 11 52 49" src="https://user-images.githubusercontent.com/1078012/195982595-d3292e3f-702c-48b2-9228-55e71beb9aa0.png">

**Comment by DanTup** [comment]
There's a VS Code quirk here with empty placeholders (I've filed https://github.com/microsoft/vscode/issues/163929). If that's fixed, this would actually not do what I expected above, it would rename the constructor to be a named constructor.

Depending on how VS Code fixes this, we may need to change this slightly to reflect that, or avoid returning valid responses from `prepare` in the cases of "adding a name to" renames (constructors, imports, etc.) so the user just gets a message saying they can't rename this.

**Comment by lukehutch** [comment]
Right now when I try renaming a class from its constructor, it creates a new named constructor, which is a very weird thing for it to do:

Before:
![image](https://github.com/Dart-Code/Dart-Code/assets/811305/a86ccaaf-49d0-4702-80be-19e608615257)

After:
![image](https://github.com/Dart-Code/Dart-Code/assets/811305/6a0264c9-d892-4d55-bcc7-6d75af4c5a10)

dart-code v3.83.20240220 (pre-release)
dart v3.4.0-99.0.dev

**Comment by DanTup** [comment]
@lukehutch yeah, I think that's currently the intended behaviour (that is, you are renaming the constructor, not the class). I'm conflicted about whether it's the right call, but it is what the server currently does.

The reason the name gets duplicated like this is because of the VS Code bug noted above - if we provide an empty string for any renames (suspect it doesn't happen only here, renaming an import to add an alias probably does the same) VS Code just grabs the word from the document and pre-fills the input with that.
